lighthouse-core/gather/driver.js

```diff
@@ -278,30 +279,41 @@ class Driver {
   sendCommand(method, ...params) {
     const timeout = this._nextProtocolTimeout;
     this._nextProtocolTimeout = DEFAULT_PROTOCOL_TIMEOUT;
-    const domainCommand = /^(\w+)\.(enable|disable)$/.exec(method);
-    if (domainCommand) {
-      const enable = domainCommand[2] === 'enable';
-      if (!this._shouldToggleDomain(domainCommand[1], enable)) {
-        return Promise.resolve();
-      }
-    }
     return new Promise(async (resolve, reject) => {
-      const asyncTimeout = setTimeout((_ => {
+      const asyncTimeout = setTimeout((() => {
         const err = new LHError(LHError.errors.PROTOCOL_TIMEOUT);
         err.message += ` Method: ${method}`;
         reject(err);
       }), timeout);
       try {
-        const result = await this._connection.sendCommand(method, ...params);
-        clearTimeout(asyncTimeout);
+        const result = await this.innerSendCommand(method, ...params);
         resolve(result);
       } catch (err) {
-        clearTimeout(asyncTimeout);
         reject(err);
+      } finally {
+        clearTimeout(asyncTimeout);
       }
     });
   }
 
+  /**
+   * Call protocol methods.
+   * @template {keyof LH.CrdpCommands} C
+   * @param {C} method
+   * @param {LH.CrdpCommands[C]['paramsType']} params
+   * @return {Promise<LH.CrdpCommands[C]['returnType']>}
+   */
+  innerSendCommand(method, ...params) {
```

Comment:
Maybe add a leading `_` and mark it with `@private` (someday tsc will support it) to make it super clear it's intended for internal use